### PR TITLE
Switch package manager to pnpm and upgrade to Storybook 8

### PR DESCRIPTION
**Pull request by ndelangen**
# What I did:

@ghengeveld requested this to be upgraded to storybook 8.x in order to use the visual regression addon.,

- I ran the storybook upgrade command
- I ran some manual migrations
- I applied some manual fixes
- Migrated the repo to use `pnpm`
- tweaked the CI so it uses `pnpm`

<details>
  <summary>📦 Published PR as canary version: <code>11.4.1--canary.987.9175126545.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install chromatic@11.4.1--canary.987.9175126545.0
  # or 
  yarn add chromatic@11.4.1--canary.987.9175126545.0
  ```
</details>

### CODE REVIEW

**Comment by thafryer** [comment]
@ndelangen What led to us using PNPM here? If you were just experimenting, I will ignore. :) Updating to Yarn 4 is very welcome though if you just wanted to update to a newer package manager.

File: node-src/tasks/verify.ts [hunk 1]
```diff
@@ -57,7 +57,7 @@ export const publishBuild = async (ctx: Context) => {
       id,
       input: {
         ...(onlyStoryFiles && { onlyStoryFiles }),
-        ...(onlyStoryNames && { onlyStoryNames: [].concat(onlyStoryNames) }),
+        ...(onlyStoryNames && { onlyStoryNames: [onlyStoryNames] }),
```
Comment (ghengeveld) [inline node-src/tasks/verify.ts:60]: @ethriel3695 This is a functional change. Why was this changed, and why is it part of this PR?

File: node-src/tasks/verify.ts [hunk 2]
```diff
@@ -249,7 +249,10 @@ export const verifyBuild = async (ctx: Context, task: Task) => {
 
   if (ctx.build.wasLimited) {
     const { account } = ctx.build.app;
-    if (account.exceededThreshold) {
+    if (!account) {
+      ctx.log.warn('No Account');
+      setExitCode(ctx, exitCodes.INVALID_OPTIONS, true);
```
Comment (ghengeveld) [inline node-src/tasks/verify.ts:254]: This wouldn't be `INVALID_OPTIONS` but something more severe, and probably not user error. Basically this should never happen so throwing an error would be the proper recourse here.

```suggestion
      throw new Error('Failed to retrieve account information');
```

File: node-src/tasks/verify.ts [hunk 3]
```diff
@@ -266,7 +269,7 @@ export const verifyBuild = async (ctx: Context, task: Task) => {
 
   transitionTo(success, true)(ctx, task);
 
-  if (list || ctx.isPublishOnly || matchesBranch(ctx.options.exitOnceUploaded)) {
+  if (list || ctx.isPublishOnly || (matchesBranch && matchesBranch(ctx.options.exitOnceUploaded))) {
```
Comment (ghengeveld) [inline node-src/tasks/verify.ts:272]:
```suggestion
  if (list || ctx.isPublishOnly || matchesBranch?.(ctx.options.exitOnceUploaded)) {
```

File: node-src/ui/tasks/verify.ts [hunk 4]
```diff
@@ -32,15 +32,17 @@ export const runOnlyFiles = (ctx: Context) => ({
...
   status: 'pending',
   title: 'Starting partial build',
-  output: `Snapshots will be limited to stories matching ${ctx.options.onlyStoryNames
-    .map((v) => `'${v}'`)
-    .join(', ')}`,
+  output: ctx.options.onlyStoryNames
+    ? `Snapshots will be limited to stories matching ${ctx.options.onlyStoryNames
+        .map((v) => `'${v}'`)
+        .join(', ')}`
+    : `Snapshots will be limited to ${ctx.onlyStoryFiles?.length} story files affected by recent changes`,
```
Comment (ghengeveld) [inline node-src/ui/tasks/verify.ts:45]: The name of this function implies that `onlyStoryNames` is defined, so falling back to `onlyStoryFiles` is not only functionally wrong but also not necessary.

**Review by ghengeveld — CHANGES_REQUESTED** [review]
This PR seems to be doing more than upgrade Storybook and the package manager. Ideally this change should not affect the CLI functionally.

File: node-src/tasks/verify.ts (hunk 1 above)
Comment (ethriel3695) [reply node-src/tasks/verify.ts:60]: @ghengeveld I changed it because TypeScript was yelling at me about it.
I'll revert.

**Comment by ethriel3695** [comment]
> This PR seems to be doing more than upgrade Storybook and the package manager. Ideally this change should not affect the CLI functionally.

@ghengeveld some changes were necessary to get tests to pass after changing to pnpm.

Other changes were due to Typescript yelling at me. I've reverted those.
